llama_index/llms/openai_utils.py

```diff
@@ -274,7 +274,10 @@ def from_openai_message_dicts(message_dicts: Sequence[dict]) -> List[ChatMessage
 
 
 def to_openai_function(pydantic_class: Type[BaseModel]) -> Dict[str, Any]:
-    """Convert pydantic class to OpenAI function."""
+    """Deprecated in favor of `to_openai_tool`.
```

Comment:
can we raise a real deprecation warning?

Reply:
https://deprecated.readthedocs.io/en/latest/tutorial.html

Reply:
done! also added deprecation warning to the similar `BaseTool.metadata.to_openai_function`